src/main/java/zmq/Ctx.java

```diff
@@ -149,6 +149,8 @@ private void destroy() throws IOException
         termMailbox.close();
 
         tag = 0xdeadbeef;
+
+        zmq.ZMQ.closePollSelector();
```

Comment:
I'm almost 2 years late to this conversation, but is there a reason that the PollSelector needs to be ThreadLocal? One thing I've read about ThreadLocal is that it makes it easy to introduce a memory leak. I think we might be seeing an example of this with the "Too many files open" problems.